packs/chef/actions/lib/omnibus.py

```diff
@@ -1,40 +1,51 @@
+import pipes
+
 import shellhelpers as shell
 
+
 class Omnibus(object):
     OMNITRUCK = 'https://www.chef.io/chef/install.sh'
 
     def __init__(self, options):
         supports = ('pre_release', 'version', 'download_path')
-        self.options = { k: options[k] for k in supports }
+        self.options = {k: options[k] for k in supports}
 
-    def build_command(self, version=None, pre_release=False, download_path=None):
-        command = "curl -sL %s | sudo bash" % self.OMNITRUCK
+    def build_command(self, version=None, pre_release=False,
+                      download_path=None):
+        command = "curl -sL %s | sudo bash" % (pipes.quote(self.OMNITRUCK))
 
-        if pre_release or version != None or download_path != None:
-          command += " -s --"
+        command_args = []
+
+        if pre_release or version or download_path:
+            command_args += ['-s', '--']
 
         if pre_release:
-          command += " -p"
+            command_args += ['-p']
 
-        if version != None:
-          command += " -v \"%s\"" % version
+        if version:
+            command_args += ['-v', version]
 
         if download_path:
-          command += " -d \"%s\"" % download_path
+            command_args += ['-v', download_path]
 
-        return command
+        command_args = [pipes.quote(arg) for arg in command_args]
+        command_args = ' '.join(command_args)
+        command = command + ' ' + command_args
 
+        return command
 
     def chef_installed(self):
         knife = "/opt/chef/bin/knife"
-        command = "test -x %s" % knife
+        command = "test -x %s" % (pipes.quote(knife))
```

Comment:
Yeah, as it stands right now, escaping is not required, but in cases like this (when dealing with security and possible injections) it's way better to be safe than sorry.

Let's say that someone makes path to knife binary user configurable via parameter and doesn't add / forgets to add pipe.quote - bam, here we have possible shell injection.

And if you use `subprocess.Popen` with a args list and not a command string, Popen will do the same for you internally (automatically escape every argument).

As far as backward compatibility goes - `pipes.quote` is still available in Python 3.3 and 3.4 so we should be fine for  a long time if you consider how slow Python 3 adoption is and how long before a new major release where this function might be removed comes out.